### PR TITLE
Update package.json latest versions and .travis.yml to build correct …

### PR DESCRIPTION
**Pull request by Dexus**
…iojs and nodeja v4 used v8 engine with modules

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,21 +2,20 @@ language: node_js
...
     - "0.10"
-#   - "0.8"   # no longer supported by iconv
-#   - "0.6"   # no longer supported by async
-    - "iojs-v1.8.1"
-    - "iojs"
-
+    - "stable"
+    - "iojs-v1"
+    - "iojs-v2"
+    - "iojs-v3"
```
Comment (msimerson) [inline .travis.yml]: I doubt we'll ever care to deploy to iojs 1, 2, or 3, and instead target node.js 4.0 (the unforked branch). So not much point in adding them here.

File: .travis.yml [hunk 2]
```diff
@@ -2,21 +2,20 @@ language: node_js
...
-#       - node_js: "0.6"
-
+addons:
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+    packages:
+      - g++-4.8
+env:
+  - "CXX=g++-4.8"
```
Comment (msimerson) [inline .travis.yml]: I think these options are no longer required.

Comment (Dexus) [inline .travis.yml]: This lines need. See https://travis-ci.org/Dexus/Haraka/builds/81483117

Comment (msimerson) [inline .travis.yml]: Okay, so put a comment there telling why: "these are required for building on node.js 4.0"

File: .travis.yml (hunk 1 above)
Comment (Dexus) [inline .travis.yml]: ok but iojs gets future updates... i dont know if only for bugs or also for new features. So I think that iojs-v3 should remain there.

File: .travis.yml [hunk 3]
```diff
@@ -2,21 +2,20 @@ language: node_js
 node_js:
     - "0.12"
     - "0.10"
-#   - "0.8"   # no longer supported by iconv
-#   - "0.6"   # no longer supported by async
```
Comment (msimerson) [inline .travis.yml]: These comments, telling **why** we don't support node 0.8 and 0.6 tend to be helpful, which is why we leave them in there.

File: package.json [hunk 4]
```diff
@@ -14,36 +14,36 @@
     "node": ">= v0.10.0"
   },
```
Comment (baudehlo) [inline package.json]: We should update this to 10.36 at least due to bad certs in the older nodes.

**Pull request merged**